### PR TITLE
[WEBDEV-764] Refactor search__global wrapper, [Release] Pugin version bump

### PR DESCRIPTION
**Pull request by naseberry**
- Refactor search global wrapper
- [Release] Pugin version bump

### CODE REVIEW

File: resources/js/bodyClass.js [hunk 1]
```diff
@@ -0,0 +1 @@
+document.body.className = ((document.body.className) ? document.body.className + ' has-js' : 'has-js');
```
Comment (jakemulley) [inline resources/js/bodyClass.js]: can we get a newline at the end of files?

File: resources/js/main.js [hunk 2]
```diff
@@ -1 +1 @@
...
...t.add("show"),this.setCookie(e,t,r))}};UK_Parliament.cookieBanner("UK_Parliament__seen_cookie_message","yes",28),UK_Parliament.search=function(){var e=document.querySelector("header"),n=e.querySelector(".uk_parliament"),i=e.querySelector("form"),a=e.querySelector("input"),t=e.querySelector(".input-group"),r=e.querySelector("button"),s=e.querySelector(".btn--toggle");function o(){window.innerWidth<=768?(s.removeAttribute("tabIndex"),i.classList.contains("open")?c():u()):(c(),s.setAttribute("tabIndex","-1"),s.setAttribute("aria-hidden","true"))}function c(){t.removeAttribute("aria-hidden"),a.removeAttribute("tabIndex"),r.removeAttribute("tabIndex")}function u(){t.setAttribute("aria-hidden","true"),a.setAttribute("tabIndex","-1"),r.setAttribute("tabIndex","-1")}s&&(o(),s.addEventListener("click",function(e){i.classList.contains("open")?(r=e,r.preventDefault(),n.classList.remove("portcullis"),i.classList.remove("open"),s.setAttribute("aria-label","open search"),u()):(t=e,t.preventDefault(),n.classList.add("portcullis"),i.classList.add("open"),s.setAttribute("aria-label","close search"),a.focus(),c());var t;var r}),window.addEventListener("orientationchange",o))},UK_Parliament.search();
```
Comment (jakemulley) [inline resources/js/main.js]: can we get a newline at the end of files?

File: test/fixtures/html/components/form/search-global.html [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+<form>
+        <div class="input-group"><label class="sr-only" for="search_box"></label><input id="search_box" maxlength="1000" name="q" pattern="^(?!.*<script|<Script).*$" aria-label="enter a search term" autocomplete="off" required="" type="search"><button aria-label="submit search" class="btn--white" type="submit" data-tracking="search"><svg aria-hidden="true" data-icon="search" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
+                    <title>Search icon</title>
```
Comment (UsmanAfzal) [inline test/fixtures/html/components/form/search-global.html]: Need to remove the data-tracking attribute

File: test/fixtures/html/components/form/search-global.html [hunk 4]
```diff
@@ -1,4 +1,25 @@
...
+            <title>Close</title>
+            <g stroke="#fff">
+                <g>
+                    <line stroke-width="2" x1="0" x2="18" y1="0" y2="18"></line>
+                </g>
+                <g>
+                    <line stroke-width="2" x1="0" x2="18" y1="18" y2="0"></line>
+                </g>
+            </g>
+        </svg></a>
```
Comment (UsmanAfzal) [inline test/fixtures/html/components/form/search-global.html]: New line to add here

File: test/fixtures/html/components/form/search-label-and-value.html [hunk 5]
```diff
@@ -1 +1,9 @@
...
+<form>
+        <div class="input-group"><label class="sr-only" for="search_box">Enter a keyword</label><input id="search_box" maxlength="1000" name="q" pattern="^(?!.*<script|<Script).*$" aria-label="enter a search term" autocomplete="off" required="" type="search" value="banana"><button aria-label="submit search" class="btn--white" type="submit" data-tracking="search"><svg aria-hidden="true" data-icon="search" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
```
Comment (UsmanAfzal) [inline test/fixtures/html/components/form/search-label-and-value.html]: data-tracking attribute needs removing here too

File: test/fixtures/html/components/form/search-label-and-value.html [hunk 6]
```diff
@@ -1 +1,9 @@
...
+                    <title>Search icon</title>
+                    <g fill="#fff">
+                        <path d="M7.5 15C3.4 15 0 11.6 0 7.5 0 3.4 3.4 0 7.5 0 11.6 0 15 3.4 15 7.5 15 11.6 11.6 15 7.5 15ZM7.5 13C10.5 13 13 10.5 13 7.5 13 4.5 10.5 2 7.5 2 4.5 2 2 4.5 2 7.5 2 10.5 4.5 13 7.5 13Z"></path>
+                        <polygon points="14.8 10.8 16.8 10.8 16.8 20.8 14.8 20.8" transform="translate(15.778175 15.778175)rotate(-45)translate(-15.778175 -15.778175)"></polygon>
+                    </g>
+                </svg></button></div>
+    </form>
```
Comment (UsmanAfzal) [inline test/fixtures/html/components/form/search-label-and-value.html]: new line

File: test/fixtures/html/components/form/search.html [hunk 7]
```diff
@@ -1,3 +1,3 @@
...
+    <div class="input-group"><label class="sr-only" for="search_box">Enter a keyword</label><input id="search_box" maxlength="1000" name="q" pattern="^(?!.*<script|<Script).*$" aria-label="enter a search term"autocomplete="off" required="" type="search"><button aria-label="submit search" class="btn--white" type="submit" data-tracking="search"><svg aria-hidden="true" data-icon="search" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg"><title>Search icon</title><g fill="#fff"><path d="M7.5 15C3.4 15 0 11.6 0 7.5 0 3.4 3.4 0 7.5 0 11.6 0 15 3.4 15 7.5 15 11.6 11.6 15 7.5 15ZM7.5 13C10.5 13 13 10.5 13 7.5 13 4.5 10.5 2 7.5 2 4.5 2 2 4.5 2 7.5 2 10.5 4.5 13 7.5 13Z"></path><polygon points="14.8 10.8 16.8 10.8 16.8 20.8 14.8 20.8" transform="translate(15.778175 15.778175)rotate(-45)translate(-15.778175 -15.778175)"></polygon></g></svg></button></div>
```
Comment (UsmanAfzal) [inline test/fixtures/html/components/form/search.html]: data-tracking attribute here too

Comment (UsmanAfzal) [inline test/fixtures/html/components/form/search.html]: There's no space after `aria-label="enter a search term"` and `autocomplete`

File: test/fixtures/html/components/section/primary-flag.html [hunk 8]
```diff
@@ -2,8 +2,7 @@
...
+            <div class="input-group"><label class="sr-only" for="search_box">Enter a keyword</label><input id="search_box" maxlength="1000" name="q" pattern="^(?!.*<script|<Script).*$" aria-label="enter a search term" autocomplete="off" required="" type="search"><button aria-label="submit search" class="btn--white" type="submit" data-tracking="search"><svg aria-hidden="true" data-icon="search" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg"><title>Search icon</title><g fill="#fff"><path d="M7.5 15C3.4 15 0 11.6 0 7.5 0 3.4 3.4 0 7.5 0 11.6 0 15 3.4 15 7.5 15 11.6 11.6 15 7.5 15ZM7.5 13C10.5 13 13 10.5 13 7.5 13 4.5 10.5 2 7.5 2 4.5 2 2 4.5 2 7.5 2 10.5 4.5 13 7.5 13Z"></path><polygon points="14.8 10.8 16.8 10.8 16.8 20.8 14.8 20.8" transform="translate(15.778175 15.778175)rotate(-45)translate(-15.778175 -15.778175)"></polygon></g></svg></button></div>
```
Comment (UsmanAfzal) [inline test/fixtures/html/components/section/primary-flag.html]: data-tracking attribute

File: view/components/form/search.dust [hunk 9]
```diff
@@ -1,19 +1,16 @@
 {! This the the search form used in the header and on search index page. !}
+
+  <form {?data.search-action}action="{data.search-action}" method="get"{/data.search-action}>
+    <div class="input-group">
+      <label class="sr-only" for="search_box">{@t key="{data.label}" /}</label>
+      <input id="search_box" maxlength="1000" name="q" pattern="^(?!.*<script|<Script).*$" aria-label="enter a search term" autocomplete="off" required="" type="search" {?data.value}value="{data.value}"{/data.value}>
+      <button aria-label="submit search" class="btn--white" type="submit" data-tracking="search">
```
Comment (UsmanAfzal) [inline view/components/form/search.dust]: data-tracking attribute

File: test/fixtures/html/components/form/search-global.html (hunk 3 above)
Comment (naseberry) [reply test/fixtures/html/components/form/search-global.html]: parliament.uk-pugin templates require updating as I lifted this attribute from the template

**Pull request merged**